lib/mix/lib/mix/tasks/help.ex

```diff
@@ -71,9 +73,13 @@ defmodule Mix.Tasks.Help do
       load_tasks()
       |> Enum.filter(&String.contains?(Mix.Task.task_name(&1), pattern))
 
-    {docs, max} = build_task_doc_list(modules)
+    aliases =
+      load_aliases()
+      |> Enum.filter(&String.contains?(elem(&1, 0), pattern))
```

Comment:
We try to avoid single pipes. I would suggest to refactor this with a `for` comprehension:

```elixir
aliases = 
  for {alias_name, _} = alias <- load_aliases(),
      String.contains?(alias_name, pattern),
      do: alias
```

Reply:
why do we try to avoid single pipes?

Reply:
@Sihui mostly because if we don't have multiple expressions, the pipeline ends up being more noise than helping readability. We know it is subjective but that's why we decided to add a "rule", so it is a clear path and we remove subjectivity. :)

Reply:
A (weak) counter argument will be using single pipe makes adding additional pipe easier. But that's for another discussion.

I don't think we should avoid `Enum.filter` b/c of avoiding single pipes though. I think `Enum.filter` expresses our intention clearly than the `for` comprehension. How about 

```elixir
Enum.filter(load_aliases, &String.contains?(elem(&1, 0), pattern))
```

A side note, we have quite a few usages of single pipes throughout the file (and the codebase), does it mean we want to replace all of them?

Reply:
Eventually we want to replace all pipes, but we try to avoid small changes like that in the codebase in order to keep the number of changes down. `Enum.filter/2` is also very much fine with me, I just don't love `elem/2` since it hides what the tuple looks like. What about:

```elixir
Enum.filter(load_aliases(), fn {name, _} -> String.contains?(name, pattern) end)
```

Reply:
good call! I not a fan of `elem` either.